### PR TITLE
Float columns

### LINKED ISSUE #83: Implement number data types
We need columns that store integers and floats. Both should be queryable via "equals" and via "between" (similar to a date).

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #83

### CODE REVIEW

**Comment by fengelniederhammer** [comment]
Do we also want a `FloatEquals`filter? What do you think @Taepper @chaoran-chen ?

File: include/silo/storage/column/float_column.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  public:
+   template <class Archive>
+   [[maybe_unused]] void serialize(Archive& archive, const unsigned int /* version */) {
+      archive& values;
+   }
+
+  private:
+   std::vector<double> values;
+
+  public:
```
Comment (JonasKellerer) [inline include/silo/storage/column/float_column.h:21]: Only a style thing: Do we want to have multiple sections of public and private in the class?

File: include/silo/storage/column/int_column.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  public:
+   template <class Archive>
+   [[maybe_unused]] void serialize(Archive& archive, const unsigned int /* version */) {
+      archive& values;
+   }
+
+  private:
+   std::vector<uint64_t> values;
+
+  public:
```
Comment (JonasKellerer) [inline include/silo/storage/column/int_column.h:21]: Same as above

**Review by JonasKellerer — APPROVED** [review]
Looks good. Only a minor formating question left.

File: include/silo/storage/column/float_column.h (hunk 1 above)
Comment (fengelniederhammer) [reply include/silo/storage/column/float_column.h:21]: In general probably not, but in this case I consider the Archive thing boiler plate, that's simply on top of the class definition. 
Then we have the rest with a private and a public section.

We could change it, but that's how we have it everywhere AFAIK.

Comment (Taepper) [reply include/silo/storage/column/float_column.h:21]: I also think the grouping by private public is a good idea in general, but other groupings can outweigh that guideline such as Fabian's mentioned consideration

**Comment by chaoran-chen** [comment]
@fengelniederhammer, re `FloatEquals`: sure, if it's not too much trouble, it's probably useful to have that as well.

**Comment by fengelniederhammer** [comment]
I implemented FloatEquals, but it somehow doesn't work yet. I don't know whether there is a mistake somewhere or whether float comparison just doesn't work.

**Comment by JonasKellerer** [comment]
I found the error with float equals:
In the metadata file we write "0.90". This is converted in `metadata_strore.cpp` line 39 through `std::stof(value)`. This produces a representation of "0.89999997615814209". However, when the query is read in `float_equals.cpp` in line 55 as "0.90000000000000002". This difference comes, because nlohmann_josn uses the a 15digit precision https://json.nlohmann.me/home/faq/#serialization-issues and stof does not. I changed it to `std::stod(value)` which should have the same precision as nlohmann. This fixes the issue for now.

**Pull request merged**